src/useradd.c

```diff
@@ -783,7 +783,7 @@ static int get_groups (char *list)
 	 * each name and look it up. A mix of numerical and string
 	 * values for group identifiers is permitted.
 	 */
-	do {
+	while (NULL != list) {
```

Comment:
Just in case it's not obvious:  We know it's not NULL in the first iteration because right above, in line 772, we've already dereferenced it.